node/src/components/chain_synchronizer.rs

```diff
@@ -110,6 +116,23 @@ impl ChainSynchronizer {
         match result {
             Ok(latest_block_header) => {
                 if latest_block_header.protocol_version() == self.chainspec.protocol_version() {
+                    if let Some(next_upgrade_activation_point) = self
+                        .maybe_next_upgrade
+                        .map(|next_upgrade| next_upgrade.era_id())
+                    {
+                        if latest_block_header.era_id() >= next_upgrade_activation_point {
```

Comment:
Shouldn't we rather stop before we've downloaded that block? Getting blocks for a future version seems unsafe.

Instead I'd make sure that we stop syncing if `latest_block_header.next_block_era_id() == next_upgrade_activation_point`, so that the new version creates the upgrade block and continues.